### PR TITLE
8259032: MappedMemorySegmentImpl#makeMappedSegment() ignores Unmapper#pagePosition

### PR DESCRIPTION
**Pull request by mcimadamore**
Hi,
this patch fixes an oversight where the address() method in FileChannelImpl.Unmapper does not take into account the value of `pagePosition` - this leads to a situation in which, effectively, the specified offset when mapping the segment is ignored by the runtime (because base address will always be aligned to some known quantity - which is OS/platform dependent).

To test this I had to break open into FileChannelImpl and ready the granularity.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8259032](https://bugs.openjdk.java.net/browse/JDK-8259032): MappedMemorySegmentImpl#makeMappedSegment() ignores Unmapper#pagePosition

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to 7a78d72474f221578c30374c3d07fa91e992c5c2

### Contributors
 * Uwe Schindler `<uschindler@openjdk.org>`

### Download
`$ git fetch https://git.openjdk.java.net/jdk16 pull/84/head:pull/84`
`$ git checkout pull/84`

### CODE REVIEW

File: test/jdk/java/foreign/TestByteBuffer.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
  * @test
  * @modules java.base/sun.nio.ch
  *          jdk.incubator.foreign/jdk.internal.foreign
- * @run testng/othervm -Dforeign.restricted=permit TestByteBuffer
+ * @run testng/othervm --illegal-access=permit -Dforeign.restricted=permit TestByteBuffer
```
Comment (AlanBateman) [inline test/jdk/java/foreign/TestByteBuffer.java]: Can you change java.base/sun.nio.ch to java.base/sun.nio.ch:+open instead? That would avoid the --illegal-access=permit.

Comment (uschindler) [reply test/jdk/java/foreign/TestByteBuffer.java]: I am not sure if a test like this is really needed.
The alignment is pageSize on Linux and some arbitrary value (65536) on Windows. If you have some test file that writes like a few bytes (1, 2, 3, 4,...) To a file and then maps it with offsets other than zero, you just have to get the first byte and compare it to offset.

Comment (uschindler) [reply test/jdk/java/foreign/TestByteBuffer.java]: In fact the new test only checks if everything is aligned like we expect, but it does not test that our mapping returns a memory segment with expected contents.

Comment (uschindler) [reply test/jdk/java/foreign/TestByteBuffer.java]: So I tend to make a simple test without reflection that writes a defined sequence of bytes to a file and then maps it with various offsets, checking that the first byte in the mapped segment is from that sequence.

Comment (mcimadamore) [reply test/jdk/java/foreign/TestByteBuffer.java]: Good suggestion!

**Comment by uschindler** [comment]
Hi,
The new test looks fine to me.
Uwe

**Review by AlanBateman — APPROVED** [review]
Good suggestion from Uwe.

**Pull request closed without merging**